src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -1517,12 +1545,13 @@ MasterIrreducibleExpressionFunctionChecker(Oid func_id, void *context)
  * tree, or the target entry sets a different column.
  */
 static bool
-TargetEntryChangesValue(TargetEntry *targetEntry, Var *column, FromExpr *joinTree)
+TargetEntryChangesValue(TargetEntry *targetEntry, Var *column, FromExpr *joinTree,
```

Comment:
I think `TargetEntryChangesValue` requires some more comments. We pass `targetEntry` and `targetColumnAttrNumber`. And, I cannot clearly follow why do we have these two cases (e.g., why not the second check is not enough for all cases?):
```C
	if (targetColumnAttrNumber != column->varattno)
	{
		/* target entry of the form SET some_other_col = <x> */
		isColumnValueChanged = false;
	}
	else if (IsA(setExpr, Var))
	{
		Var *newValue = (Var *) setExpr;
		if (newValue->varattno == column->varattno)
		{
			/* target entry of the form SET col = table.col */
			isColumnValueChanged = false;
		}
	}
```

Reply:
You mean why is 

```
		if (newValue->varattno == column->varattno)
		{
			/* target entry of the form SET col = table.col */
			isColumnValueChanged = false;
		}
```

not enough?

If that is the question, that check is seeing if we have something like `UPDATE test SET c=c`, in which case we count that as `column value not changed`.